test/transformer.spec.js

```diff
@@ -324,6 +341,125 @@ describe( 'transformer', () => {
+                        'a spreadsheet.csv': 'hallo spaceboy/little wonder.csv',
+                        'a link.xml': 'hallo spaceboy/wishful beginnings.xml',
+                    },
+                },
+                {
+                    description: 'escaped keys',
+                    media: {
+                        'first%20image.jpg': 'hallo spaceboy/spiders from mars.jpg',
+                        'a%20song.mp3': 'hallo spaceboy/space oddity.mp3',
+                        'some%20video.mp4': 'hallo spaceboy/a small plot of land.mp4',
+                        'another%20image.png': 'hallo spaceboy/under pressure.png',
+                        'an%20instance.xml': 'hallo spaceboy/golden years.xml',
+                        'a%20spreadsheet.csv': 'hallo spaceboy/little wonder.csv',
+                        'a%20link.xml': 'hallo spaceboy/wishful beginnings.xml',
+                    },
+                },
+                {
+                    description: 'escaped values',
+                    media: {
+                        'first image.jpg': 'hallo%20spaceboy/spiders%20from%20mars.jpg',
+                        'a song.mp3': 'hallo%20spaceboy/space%20oddity.mp3',
+                        'some video.mp4': 'hallo%20spaceboy/a%20small%20plot%20of%20land.mp4',
+                        'another image.png': 'hallo%20spaceboy/under%20pressure.png',
+                        'an instance.xml': 'hallo%20spaceboy/golden%20years.xml',
+                        'a spreadsheet.csv': 'hallo%20spaceboy/little%20wonder.csv',
+                        'a link.xml': 'hallo%20spaceboy/wishful%20beginnings.xml',
+                    },
+                },
+                {
+                    description: 'escaped keys and values',
+                    media: {
+                        'first%20image.jpg': 'hallo%20spaceboy/spiders%20from%20mars.jpg',
+                        'a%20song.mp3': 'hallo%20spaceboy/space%20oddity.mp3',
+                        'some%20video.mp4': 'hallo%20spaceboy/a%20small%20plot%20of%20land.mp4',
+                        'another%20image.png': 'hallo%20spaceboy/under%20pressure.png',
+                        'an%20instance.xml': 'hallo%20spaceboy/golden%20years.xml',
+                        'a%20spreadsheet.csv': 'hallo%20spaceboy/little%20wonder.csv',
+                        'a%20link.xml': 'hallo%20spaceboy/wishful%20beginnings.xml',
+                    },
+                },
+            ];
+
+            before( done => {
+                Promise.all( mediaMaps.map( ( { media } ) => (
+                    transformer.transform( {
+                        xform,
+                        media,
+                    } )
+                ) ) ).then( transformed => {
+                    results = transformed;
+                    done();
+                }, done );
+            } );
+
+            mediaMaps.forEach( ( { description }, index ) => {
```

Comment:
Oooo I really didn't know how to feel about the `forEach` being used to create a "table test" here, but the more I think about it the more it feels like the best way to test this. Just thinking out loud.

Does the different possible versions of the media map account for different Open Rosa servers?

Reply:
> Oooo I really didn't know how to feel about the forEach being used to create a "table test" here

I wasn't especially keen on it either. Other test frameworks provide a `describe.each` or `it.each` to make this approach more clear. Some changes that might help improve readability/clarity:

1. Inline `mediaMaps` at the `forEach` call site
2. Move `before` into the describe block to accommodate that
3. Use only one fixture (unescaped) and iterate over the escaping permutations programmatically instead (but this may also obscure the intent of the tests more than we'd want)

> Does the different possible versions of the media map account for different Open Rosa servers?

This is more of a guard in case behavior changes, and a general Robustness Principle check, which I think is warranted given the nature of the original bug. I believe Central currently sends unescaped values, but once this and enketo/enketo-express#291 are released, nothing should break if that changes in Central.